lib/vault/api/app_id.rb

```diff
@@ -0,0 +1,31 @@
+require "json"
+
+require_relative "secret"
+require_relative "../client"
+
+module Vault
+  class Client
+    # A proxy to the {AppId} methods.
+    # @return [AppId]
+    def app_id
+      @app_id ||= AppId.new(self)
+    end
+  end
+
+  class AppId < Request
+    # Create an authentication token.
+    #
+    # @example
+    #   Vault.app_id.login('foo', 'bar') #=> #<Vault::Secret lease_id="">
+    #
+    # @param [String] app_id
+    # @param [String] user_id
+    #
+    # @return [Secret]
+    def login(app_id, user_id)
```

Comment:
Actually, after a bit more consideration, it _shouldn't_ persist to disk, but it should store the resulting token in memory for future requests I think.